python/helpers.py

```diff
@@ -0,0 +1,170 @@
+import base64
+
+from pymongo import MongoClient
+from pymongo.encryption_options import AutoEncryptionOpts
+from pymongo.encryption import ClientEncryption
+from bson.codec_options import CodecOptions
+from bson.binary import Binary, STANDARD, UUID
+from bson import json_util
+
+
+def read_master_key(path="./master-key.txt"):
+    return Binary(open(path, "rb").read(96))
+
+
+def create_json_schema(data_key=None):
+    if data_key is None:
+        raise ValueError("data_key must be provided")
+
+    patient_schema = """
+    {{
+        "medicalRecords.patients": {{
+            "bsonType": "object",
+            "encryptMetadata": {{
+                "keyId": [
+                    {{
+                        "$binary": {{
+                            "base64": "{data_key}",
+                            "subType": "04"
+                        }}
+                    }}
+                ]
+            }},
+            "properties": {{
+                "insurance": {{
+                    "bsonType": "object",
+                    "properties": {{
+                        "policyNumber": {{
+                            "encrypt": {{
+                                "bsonType": "int",
+                                "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic"
+                            }}
+                        }}
+                    }}
+                }},
+                "medicalRecords": {{
+                    "encrypt": {{
+                        "bsonType": "array",
+                        "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
+                    }}
+                }},
+                "bloodType": {{
+                    "encrypt": {{
+                        "bsonType": "string",
+                        "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
+                    }}
+                }},
+                "ssn": {{
+                    "encrypt": {{
+                        "bsonType": "int",
+                        "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic"
+                    }}
+                }}
+            }}
+        }}
+    }}
+    """.format(data_key=data_key)
+
+    return json_util.loads(patient_schema)
+
+
+class ClientBuilder:
+    def __init__(self,
+                 kms_providers=None,
+                 key_alt_name="demo-data-key",
+                 key_db="encryption",
+                 key_coll="__keyVault",
+                 schema=None,
+                 connection_string="mongodb://localhost:27017",
+                 mongocryptd_bypass_spawn=False,
+                 mongocryptd_spawn_path="mongocryptd"):
+        """
```

Comment:
Nitpick: this comment block should probably be located above the class declaration rather than within the `__init__` function